python/pyiceberg/io/pyarrow.py

```diff
@@ -230,6 +240,14 @@ def _get_fs(self, scheme: str) -> FileSystem:
         else:
             raise ValueError(f"Unrecognized filesystem type in URI: {scheme}")
 
+    def _get_file(self, location: str) -> PyArrowFile:
+        scheme, path = self.parse_location(location)
+        fs = self._get_fs(scheme)
+
+        buffer_size = self.properties.get(BUFFER_SIZE)
```

Comment:
Just a nit, I feel it's a bit more readable if we return the 1 MB default buffer size here and then pass that through. Then buffer size is a required integer for PyArrowFile and the constructor logic is simpler imo.

Reply:
Makes a lot of sense, let me update that! 👍🏻